libs/community/langchain_community/vectorstores/cratedb/base.py

```diff
@@ -324,14 +324,21 @@ def _query_collection_multi(
             results: List[Any] = (
                 session.query(  # type: ignore[attr-defined]
                     self.EmbeddingStore,
-                    # FIXME: Using `_score` is definitively the wrong choice.
-                    #        - https://github.com/crate-workbench/langchain/issues/19
-                    #        - https://github.com/crate/crate/issues/15835
                     # TODO: Original pgvector code uses `self.distance_strategy`.
                     #       CrateDB currently only supports EUCLIDEAN.
                     #       self.distance_strategy(embedding).label("distance")
-                    sqlalchemy.literal_column(
-                        f"{self.EmbeddingStore.__tablename__}._score"
+                    sqlalchemy.func.vector_similarity(
+                        self.EmbeddingStore.embedding,
+                        # TODO: Just reference the `embedding` symbol here, don't
+                        #       serialize its value prematurely.
+                        #       https://github.com/crate/crate/issues/16912
+                        #
+                        # Until that got fixed, marshal the arguments to
+                        # `vector_similarity()` manually, in order to work around
+                        # this edge case bug. We don't need to use JSON marshalling,
+                        # because Python's string representation of a list is just
+                        # right.
+                        sqlalchemy.text(str(embedding)),
```

Comment:
See also:
- https://github.com/crate/crate/issues/16912

Reply:
> I suppose we need a fix of https://github.com/crate/crate/issues/16912 before proceeding?

Not quite. The snippet above uses a workaround, by marshalling the `embedding` argument to `vector_similarity()` manually, instead of submitting it as an SQL parameter. In this spirit, we can proceed, and submit an improvement later.

Reply:
If you are ready for a bigger review, I am fine with toggling it ready :smile: